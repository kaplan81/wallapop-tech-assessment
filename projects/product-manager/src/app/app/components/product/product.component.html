<div class="product">
  <mat-card *ngIf="product">
    <mat-card-header>
      <mat-card-title-group>
        <mat-card-title>{{ product.title }}</mat-card-title>
        <mat-card-subtitle>{{ product.email }}</mat-card-subtitle>
        <mat-card-subtitle>{{ product.price }}&euro;</mat-card-subtitle>
        <img mat-card-sm-image [src]="product.image" />
      </mat-card-title-group>
    </mat-card-header>
    <mat-icon
      fontIcon="star"
      [class.selected]="product.isFavourite"
      (click)="updateFavourites(product.email, product.isFavourite)"
    ></mat-icon>
    <mat-card-content>
      {{ product.description }}
    </mat-card-content>
  </mat-card>
</div>
