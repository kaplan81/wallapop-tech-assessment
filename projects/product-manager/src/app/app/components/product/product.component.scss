.product {
  mat-card {
    margin: 0 30px;
    mat-icon {
      margin: 0 15px 5px 15px;
      opacity: 25%;
      cursor: pointer;
      &.selected {
        opacity: 100%;
      }
    }
  }
}
