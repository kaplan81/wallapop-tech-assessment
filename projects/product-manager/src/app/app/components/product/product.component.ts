import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MatCardModule } from '@angular/material/card';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [MatCardModule],
  standalone: true,
  styleUrls: ['./product.component.scss'],
  selector: 'mng-product',
  templateUrl: './product.component.html',
})
export class ProductComponent {}
