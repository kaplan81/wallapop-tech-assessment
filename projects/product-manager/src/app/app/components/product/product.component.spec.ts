import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProductComponent } from './product.component';

describe('ProductComponent', () => {
  let fixture: ComponentFixture<ProductComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ProductComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(ProductComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
