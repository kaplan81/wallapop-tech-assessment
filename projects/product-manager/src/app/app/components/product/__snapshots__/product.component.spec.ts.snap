// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductComponent should match snapshot 1`] = `
<mng-product
  __ngContext__="0"
  closeDialog={[Function EventEmitter_]}
  product="null"
>
  <div
    class="product"
  >
  </div>
</mng-product>
`;
